### PR TITLE
Urlhandling from derivgo

### PR DESCRIPTION
**Pull request by ashraf-deriv**
Changes:

-   ...

## Type of change

-   [ ] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/pages/help-centre/_utility.js [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
-    return '/help-centre/' + category.replace(/\s/g, '-').toLowerCase() + '/#' + label
+export const convertToHash = (category, label, qparam) => {
+    if (qparam) {
+        return (
+            '/help-centre/' +
+            category.replace(/\s/g, '-').toLowerCase() +
+            '/?platform=' +
+            qparam +
+            '#' +
+            label
```
Comment (suthesan-deriv) [inline src/pages/help-centre/_utility.js:9]: Can we have this a template literal and abstract out the replace function?

File: src/pages/help-centre/_utility.js [hunk 2]
```diff
@@ -1,5 +1,16 @@
...
+        )
+    } else {
+        return '/help-centre/' + category.replace(/\s/g, '-').toLowerCase() + '/#' + label
```
Comment (suthesan-deriv) [inline src/pages/help-centre/_utility.js:12]: same here

File: src/components/layout/layout.js [hunk 3]
```diff
@@ -131,7 +131,15 @@ const Layout = ({
     const [gtm_data, setGTMData] = useGTMData()
 
     const is_static = type === 'static'
-
+    const queryParams = new URLSearchParams(window.location.search)
+    const term = queryParams.get('platform')
```
Comment (sean-binary) [inline src/components/layout/layout.js:135]: I think calling it `page` would be more descriptive than `term`.

File: src/pages/help-centre/_utility.js (hunk 1 above)
Comment (ashraf-deriv) [reply src/pages/help-centre/_utility.js:9]: Sure, That would be better. Let me try.

File: src/components/layout/layout.js (hunk 3 above)
Comment (ashraf-deriv) [reply src/components/layout/layout.js:135]: Ok , we can update. Thanks

File: src/pages/help-centre/_article.tsx [hunk 4]
```diff
@@ -36,6 +36,11 @@ const TabWrapper = styled.div`
 const ContactContainer = styled.div`
     margin-top: 8rem;
 `
+
+const queryParams = new URLSearchParams(window.location.search)
+const platform_name = queryParams.get('platform')
+const platform_list = ['derivgo', 'p2p']
+const platform = platform_list.includes(platform_name) ? platform_name : ''
```
Comment (sean-binary) [inline src/pages/help-centre/_article.tsx:43]: We can make this a utility since we use it in multiple places.

Comment (ashraf-deriv) [reply src/pages/help-centre/_article.tsx]: Yep, it would be good. Thank you

File: src/pages/help-centre/_article.tsx [hunk 5]
```diff
@@ -56,7 +57,7 @@ export const Article = ({
             />
             <Container align="left" justify="flex-start" direction="column">
                 <StyledLink
-                    to="/help-centre/"
+                    to={platform ? '/help-centre/?platform=' + platform : '/help-centre/'}
```
Comment (sean-binary) [inline src/pages/help-centre/_article.tsx:60]:
```suggestion
                    to={platform ? `/help-centre/?platform=${platform}` : '/help-centre/'}
```

File: src/components/layout/layout.js [hunk 6]
```diff
@@ -131,7 +130,13 @@ const Layout = ({
...
 
     const is_static = type === 'static'
-
+    if (queryParamData()) {
+        return (
+            <Main margin_top={'0'} is_static={is_static}>
+                {children}
+            </Main>
+        )
+    }
```
Comment (sean-binary) [inline src/components/layout/layout.js:139]: Let's move this to after the switch case statements.

File: src/pages/help-centre/_article.tsx [hunk 7]
```diff
@@ -36,6 +36,7 @@ const TabWrapper = styled.div`
 const ContactContainer = styled.div`
     margin-top: 8rem;
 `
+const platform = queryParamData() ? queryParamData() : ''
```
Comment (sean-binary) [inline src/pages/help-centre/_article.tsx:39]: Isn't this handled by the else case in `queryParamData`?

Comment (ashraf-deriv) [reply src/pages/help-centre/_article.tsx:39]: That's right will remove the condition, Thanks Sean.

**Comment by sean-binary** [comment]
Also please help to resolve the conflicts by pulling the upstream. `git pull upstream master`

**Comment by ashraf-deriv** [comment]
>

**Review by sean-binary — APPROVED** [review]
LGTM 👍

**Pull request merged**